src/main/java/com/worksap/fig/lang/Seq.java

```diff
@@ -404,4 +431,189 @@ default int findLastIndex(Predicate<T> condition) {
     default Seq<T> subSeq(int fromIndex, int toIndex) {
         return new SeqImpl<>(this.subList(fromIndex, toIndex));
     }
+
+    /**
+     * Remove elements which satisfy the condition, resulting a new seq without changing the original one.
+     *
+     * @return The new seq after the change
+     */
+    default Seq<T> removeIf(BiPredicate<T, Integer> condition) {
+        Objects.requireNonNull(condition);
+        Seq<T> copy = new SeqImpl<>();
+        this.forEachWithIndex((e, i) -> {
+            if (!condition.test(e, i))
+                copy.add(e);
+        });
+        return copy;
+    }
+
+    /**
+     * Remove elements which satisfy the condition at the original seq.
+     *
+     * @return The seq itself after the change
+     */
+    default Seq<T> removeIf$(BiPredicate<T, Integer> condition) {
+        Objects.requireNonNull(condition);
+        final Iterator<T> each = iterator();
+        int index = 0;
+        while (each.hasNext()) {
+            if (condition.test(each.next(), index)) {
+                each.remove();
+            }
+            index++;
+        }
+        return this;
+    }
+
+    /**
+     * Get elements which satisfy the condition, resulting a new seq without changing the original one.
+     *
+     * @return The new seq
+     */
+    default Seq<T> getIf(Predicate<T> condition) {
```

Comment:
We'd better unify the name, how about `filter`? I think Java 8 already use this word.